pkg/datastore/kubernetes/kubernetes.go

```diff
@@ -87,6 +71,56 @@ func NewDatastore(thisClusterID string, stopCh <-chan struct{}) (datastore.Datas
 	}, nil
 }
 
+func getSubmarinerClientset(host string, k8sSpec *datastoreSpecification) (*submarinerClientset.Clientset, error) {
+	// We need to check whether we need the provided CA (for a secure connection)
+	// For an insecure connection, both cases are equivalent, but we’ll never get the X.509 error
+	// First, try without specifying anything (using the system trust store)
+	clientset, err := getAndCheckSubmarinerClientset(host, false, k8sSpec)
```

Comment:
I'd cache this once learned, you could use a bool variable, initialized to "false"

clientset, err := getAndCheckSubmarinerClientset(host, usePrivateCA, k8sSpec) 

if ... failed CA ... {
     clientset, err = getAndCheckSubmarinerClientset(host, !usePrivateCA, k8sSpec)
     if err == nil {
           usePrivateCA = !usePrivateCA
     }
}

Reply:
I'm thinking of failure modes where we get disconnected from the API, may be system overloaded... we every time we try to connect again we would be imposing extra retries even if we already know what's the right certificate chain to use.

Reply:
Done.